pkg/apis/pipeline/v1beta1/taskref_conversion.go

```diff
@@ -49,6 +49,7 @@ func (tr *TaskRef) ConvertFrom(ctx context.Context, source v1.TaskRef) {
 // default and it will be in beta before the stored version of CRD getting swapped to v1.
 func (tr TaskRef) convertBundleToResolver(sink *v1.TaskRef) {
 	if tr.Bundle != "" {
+		sink.Kind = ""
```

Comment:
Remove the original "kind" attribute to stay consistent with the behavior elsewhere: once a resolver is used, "kind" is no longer necessary.